src/test/unit/Guardian.t.sol

```diff
@@ -0,0 +1,718 @@
+        pausedVar1 = pausedVar1 && pausedVar1_;
+        pausedVar2 = pausedVar2 && pausedVar2_;
+        emit PauseUpdate(msg.sender, pausedVar1, pausedVar2);
+    }
+
+    function unPause() external override {
+        require(block.timestamp > pauseTimestamp + 30 days, "G_UP: Cannot unPause");
+        if (pausedVar1 || pausedVar2) {
+            pausedVar1 = false;
+            pausedVar2 = false;
+            emit PauseUpdate(msg.sender, false, false);
+        }
+    }
+
+    function resetPauseVars() external onlyOwner {
+        pausedVar1 = false;
+        pausedVar2 = false;
+    }
+}
+
+contract MainRegistryMockup is MainRegistryGuardian {
+    uint256 public storedIncrement;
+
+    event Deposit(uint256 indexed amount, uint256 indexed increment);
+    event Withdraw(uint256 indexed amount, uint256 indexed increment);
+
+    function depositGuarded(uint256 amount) external whenDepositNotPaused {
+        storedIncrement += 1;
+        emit Deposit(amount, storedIncrement);
+    }
+
+    function withdrawUnguarded(uint256 amount) external {
+        storedIncrement += 1;
+        emit Withdraw(amount, storedIncrement);
+    }
+
+    function withdrawGuarded(uint256 amount) external whenWithdrawNotPaused {
+        storedIncrement += 1;
+        emit Withdraw(amount, storedIncrement);
+    }
+
+    function reset() external onlyOwner {
+        storedIncrement = 0;
+    }
+
+    function resetPauseVars() external onlyOwner {
+        withdrawPaused = false;
+        depositPaused = false;
+    }
+}
+
+contract FactoryMockup is FactoryGuardian {
+    uint256 public storedIncrement;
+
+    event Create(uint256 indexed id, uint256 indexed increment);
+    event Liquidate(uint256 indexed id, uint256 indexed increment);
+
+    function createGuarded(uint256 id) external whenCreateNotPaused {
+        storedIncrement += 1;
+        emit Create(id, storedIncrement);
+    }
+
+    function liquidateUnguarded(uint256 id) external {
+        storedIncrement += 1;
+        emit Liquidate(id, storedIncrement);
+    }
+
+    function liquidateGuarded(uint256 id) external whenLiquidateNotPaused {
+        storedIncrement += 1;
+        emit Liquidate(id, storedIncrement);
+    }
+
+    function reset() external onlyOwner {
+        storedIncrement = 0;
+    }
+
+    function resetPauseVars() external onlyOwner {
+        createPaused = false;
+        liquidatePaused = false;
+    }
+}
+
+contract BaseGuardianUnitTest is Test {
+    using stdStorage for StdStorage;
+
+    BaseGuardianPossibleExtension baseGuardian;
+    address guardian = address(1);
+    address owner = address(2);
+
+    constructor() {
+        vm.startPrank(owner);
+        baseGuardian = new BaseGuardianPossibleExtension();
+        baseGuardian.changeGuardian(guardian);
+        vm.stopPrank();
+    }
+
+    function setUp() public virtual {
+        // Reset the variables
+        vm.startPrank(owner);
+        baseGuardian.resetPauseVars();
```

Comment:
all tests start with a vm state after the constructor, thus the vars are already on false etc